ui/src/shared/components/time_series/TimeSeries.tsx

```diff
@@ -265,13 +267,14 @@ class TimeSeries extends Component<Props, State> {
     return null
   }
 
-  private isPropsDifferent(nextProps: Props) {
-    const isSourceDifferent = !_.isEqual(this.props.source, nextProps.source)
+  private isPropsDifferent(prevProps: Props) {
+    const isSourceDifferent = !_.isEqual(this.props.source, prevProps.source)
 
     return (
-      this.props.inView !== nextProps.inView ||
-      !!this.queryDifference(this.props.queries, nextProps.queries).length ||
-      !_.isEqual(this.props.templates, nextProps.templates) ||
+      this.props.manualRefresh !== prevProps.manualRefresh ||
+      this.props.inView !== prevProps.inView ||
+      !!this.queryDifference(this.props.queries, prevProps.queries).length ||
+      !_.isEqual(this.props.templates, prevProps.templates) ||
```

Comment:
nice! and just for consistency and readability, how would you feel about putting all of these comparisons into variables like `isSourceDifferent`?

Reply:
hm I like the idea but I'm not really sure which one is more readable, so I'm going to leave it for now